api/cpp/docs/mcu/stm32/generic.md

```diff
@@ -0,0 +1,41 @@
+<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
+
+# Generic Instructions for Slint on STM32 MCUs
+
+The following instructions outline a rough, general path how to get started on an STM32 MCU
+with STM32 Cube tooling and Slint. Successful completion requires experience with STM32CubeMX
+as well as the peripherals of our board.
+
+1. Make sure to install all the <project:../stm32.md#prerequisites>.
+2. Start a new project with STM32CubeMX:
+   - Select your base board.
+   - Enable all peripherals needed. This includes LTDC and typically the FMC to be able
+     to place the framebuffers in RAM.
+   - Select CMake in the code generator options.
+3. General the CMake project and skeleton code.
+4. In the STM32 VS Code Extension, choose the command to import a CMake project.
+5. In STM32CubeMX select the STM32Cube BSP that matches your board and install it.
+6. Copy the BSP drivers into your project's source tree and modify `CMakeLists.txt` to
+   add them to the build.
+7. Add C++ support to the generated CMake project by adding `CXX` to the `LANGUAGES` option
+   of the `project` command.
+8. Download and extract <https://github.com/slint-ui/slint/releases/latest/download/Slint-cpp-nightly-thumbv7em-none-eabihf.tar.gz>
+9. Set `CMAKE_PREFIX_PATH` to the extracted directory.
+10. Adjust your `CMakeLists.txt` for Slint use:
+    1.  Add a `find_package(Slint)` call.
+    2.  Enable resource embedding by adding `set(DEFAULT_SLINT_EMBED_RESOURCES embed-for-software-renderer)`.
```

Comment:
Why is this step needed. The default in the Slint package for MCU should be set to that.

Reply:
This is a cache variable. It's not set when building our binaries and even then it would just end up in the temporary cache there, while the application's final cache is where the value will be taken from.

We could implement it so that this is done similar to the style:

```cmake
set(SLINT_STYLE @_SLINT_STYLE@ CACHE STRING "The Slint widget style")
set_property(GLOBAL PROPERTY SLINT_STYLE ${SLINT_STYLE})
```

^^^ So this also done for the resource default. But that's not implemented right now.

Reply:
ah ok, true. i guess we should implement that.
why recommend the use of DEFAULT_SLINT_EMBED_RESOURCES instead of setting the taget property as it is meant to be used? From my understanding, DEFAULT_SLINT_EMBED_RESOURCES is used when configuring with the command line.

Reply:
I don't know TBH. I think it's slightly less verbose, but I can change it to use the target property :)